pkg/operator/controller/certificate/default_cert.go

```diff
@@ -17,67 +17,73 @@ import (
 
 // ensureDefaultCertificateForIngress creates or deletes an operator-generated
 // default certificate for a given IngressController as appropriate.  Returns true
-// if it creates, updates, or deletes the secret for the certificate, false
+// if it creates or deletes the secret for the certificate, false
 // otherwise.
 func (r *reconciler) ensureDefaultCertificateForIngress(caSecret *corev1.Secret, namespace string, deploymentRef metav1.OwnerReference, ci *operatorv1.IngressController) (bool, error) {
 	ca, err := crypto.GetCAFromBytes(caSecret.Data["tls.crt"], caSecret.Data["tls.key"])
 	if err != nil {
 		return false, fmt.Errorf("failed to get CA from secret %s/%s: %v", caSecret.Namespace, caSecret.Name, err)
 	}
-	desired, err := desiredRouterDefaultCertificateSecret(ca, namespace, deploymentRef, ci)
+	wantCert, desired, err := desiredRouterDefaultCertificateSecret(ca, namespace, deploymentRef, ci)
 	if err != nil {
 		return false, err
+	} else if !wantCert {
+		// If the opeartor generated certificate is not being used, ensure that the ingress controller's
+		// Spec.DefaultCertificate secret exists before deleting the operator generated secret.
```

Comment:
```suggestion
		// If the operator-generated certificate is not being used, ensure that the ingress controller's
		// Spec.DefaultCertificate secret exists before deleting the operator generated secret.
```
Maybe link to the Bugzilla report in the comment.

`ensureDefaultCertificateForIngress` manages the secret in the `openshift-ingress` namespace.  In the scenario described in the Bugzilla report, the auth operator breaks because the certificate is removed from the secret the `openshift-config-managed` namespace, right?  The latter secret is managed by `ensureRouterCertsGlobalSecret` in the certificate-publisher controller, and it currently doesn't publish the secret unless it is referenced by an ingresscontroller, so I think changes will be required to that logic as well.

Reply:
Ah, I naively though that the `router-certs-default` secret being deleted was the only crux of the bug. I will make the changes in the certificate-publisher as well.

Reply:
I've updated the certificate publisher controller. @Miciah PTAL when you get the chance.